### PR TITLE
feature: bcs-ops supports apiserver HA

### PR DESCRIPTION
**Pull request by Maclon9573**
1. [APIServer高可用方案 - bcs-apiserver-proxy#2344](https://github.com/TencentBlueKing/bk-bcs/issues/2344)
2. 修复一些bug

### CODE REVIEW

File: bcs-ops/install_master.sh [hunk 1]
```diff
@@ -69,6 +68,23 @@ if [[ -z ${MASTER_JOIN_CMD:-} ]]; then
...
+      "${ROOT_DIR}"/k8s/operate_bap apply
+    fi
+  fi
 else
-  kubeadm join --config="${ROOT_DIR}/kubeadm-config" -v 11
+  kubeadm join --config="${ROOT_DIR}/kubeadm-config" -v 11 \
+  || utils::log "FATAL" "failed to join master ${LAN_IP}"
+  if [ ${ENABLE_APISERVER_HA} == "true" ]; then
+    [ -z ${VIP} ] && utils::log "ERROR" "apiserver HA is enabled but VIP is not set"
+    "${ROOT_DIR}"/system/config_bc.s_dns -u "${LAN_IP}" k8s-api.bcs.local
```
Comment (bingoct) [inline bcs-ops/install_master.sh]: "${ROOT_DIR}"/system/config_bc.s_dns 多了一个.

File: bcs-ops/install_master.sh [hunk 2]
```diff
@@ -69,6 +68,23 @@ if [[ -z ${MASTER_JOIN_CMD:-} ]]; then
...
+    if [ ${APISERVER_HA_MODE} == "kube-vip" ]; then
+      "${ROOT_DIR}"/k8s/operate_kube_vip apply
+    elif [ ${APISERVER_HA_MODE} == "bcs-apiserver-proxy" ]; then
+      "${ROOT_DIR}"/k8s/operate_bap apply
+    fi
+  fi
 else
-  kubeadm join --config="${ROOT_DIR}/kubeadm-config" -v 11
+  kubeadm join --config="${ROOT_DIR}/kubeadm-config" -v 11 \
+  || utils::log "FATAL" "failed to join master ${LAN_IP}"
```
Comment (bingoct) [inline bcs-ops/install_master.sh]: 建议： ${LAN_IP}  failed to join master: ${K8S_CTRL_IP}

File: bcs-ops/install_node.sh [hunk 3]
```diff
@@ -59,16 +59,59 @@ safe_source() {
...
+      if ! command -v ctr &>/dev/null; then
+        utils::log "ERROR" "containerd client: ctr is not found"
+      fi
+      if ctr i pull --hosts-dir "/etc/containerd/certs.d" "${bap_image}"; then
+        if ! ctr run --rm --mount type=bind,src=${PROXY_TOOL_PATH},dst=/tmp,options=rbind:rw "${bap_image}" \
+          bap-copy."$(date +%s)" /bin/cp -f /data/bcs/bcs-apiserver-proxy/apiserver-proxy-tools /tmp/; then
+          utils::log "ERROR" "containerd fail to run ${bap_image}"
+        fi
+      else
+        utils::log "ERROR" "pull helm image failed"
```
Comment (bingoct) [inline bcs-ops/install_node.sh]: pull ${bap_image} image failed"

**Review by bingoct — COMMENTED** [review]
something need fix

File: bcs-ops/install_master.sh (hunk 1 above)
Comment (Maclon9573) [reply bcs-ops/install_master.sh]: 已修复

File: bcs-ops/install_master.sh (hunk 2 above)
Comment (Maclon9573) [reply bcs-ops/install_master.sh]: 已修复

File: bcs-ops/install_node.sh (hunk 3 above)
Comment (Maclon9573) [reply bcs-ops/install_node.sh]: 已修复

**Review by bingoct — APPROVED** [review]

**Review by bingoct — APPROVED** [review]

**Review by bingoct — APPROVED** [review]

File: bcs-ops/install_node.sh [hunk 4]
```diff
@@ -78,3 +121,11 @@ kubeadm --config="${ROOT_DIR}/kubeadm-config" config images pull \
   || utils::log "FATAL" "fail to pull k8s image"
 
 kubeadm join --config="${ROOT_DIR}/kubeadm-config" -v 11
+if [ ${ENABLE_APISERVER_HA} == "true" ]; then
+  if [ ${APISERVER_HA_MODE} == "bcs-apiserver-proxy" ]; then
+    init_bap_rule
```
Comment (denglouping) [inline bcs-ops/install_node.sh:126]: 需要判断执行结果

**Review by denglouping — APPROVED** [review]

**Review by abstractmj — APPROVED** [review]

**Pull request merged**